### PR TITLE
3.18.y from Raspberry Pi 2

### PR DESCRIPTION
**Pull request by f1vefour**
Should be ready to build.

### CODE REVIEW

**Comment by f1vefour** [comment]
I built the kernel commit by commit to make certain everything was okay. Let me know what needs fixing up if you see anything.

**Comment by CurlyMoo** [comment]
Currently testing...

**Comment by CurlyMoo** [comment]
First feedback:

1. Can we rename the folder from rpi-3.18.y to rpi2-3-18.y
2. Then add `rpi2*` to the list in the `zfs` hooks.
3. In the `config` add `config_deb_pkgname=linux-image-3.18.5-bcm2836`
4. In the `config` add `config_deb_ctrlextra=Conflicts: xbian-package-kernel`

**Comment by CurlyMoo** [comment]
5. Can we make the default cross_compiler `arm-linux-gnueabi-` like the other packages.

**Comment by f1vefour** [comment]
Sure, I just based the configuration off 3.17.y which doesn't have arm-linux-gnueabi- in the configuration.

I will make those changes right quick.

**Comment by CurlyMoo** [comment]
6) Can you add a `boot` folder `build/content-tpl`
7) Add a control file structure in `post-deb.d` for the rpi2 for zfs and the headers.

**Comment by f1vefour** [comment]
Question
> 6) Can you add a boot folder build/content-tpl

Which do you mean? 

1.) build/content-tpl/boot
2.) build/rpi2/content-tpl/boot
3.) build/rpi2-3.18.y/content-tpl/boot

The rest is done, but I didn't understand this request.

**Comment by CurlyMoo** [comment]
2.) shouldn't exist.

In 1.)

**Comment by f1vefour** [comment]
You want me to close this and do another pull request?

**Comment by CurlyMoo** [comment]
No, just continue :smile:

**Comment by f1vefour** [comment]
I did everything you lined out.

File: build/rpi2-3-18.y/config [hunk 1]
```diff
@@ -0,0 +1,7 @@
+config_source_configure=
+config_build_env=CROSS_COMPILE=arm-linux-gnueabihf- ARCH=arm
+config_source_branch=rpi2-3-18.y
+config_source_addr=git://github.com/raspberrypi/linux.git
+config_deb_depends=xbian-package-firmware (>=1.4.15), rng-tools, xbian-package-initramfs-tools (>= 1.3.3)
+config_deb_pkgname=linux-image-3.18.5-bcm2836
```
Comment (CurlyMoo) [inline build/rpi2-3-18.y/config]: Can you rename `linux-image-3.18.5-bcm2836` to `linux-image-bcm2836`

File: build/rpi2-3-18.y/config [hunk 2]
```diff
@@ -0,0 +1,7 @@
+config_source_configure=
+config_build_env=CROSS_COMPILE=arm-linux-gnueabihf- ARCH=arm
+config_source_branch=rpi2-3-18.y
```
Comment (CurlyMoo) [inline build/rpi2-3-18.y/config]: My comment wasn't about this line. So change `rpi2-3-18.y` back to `rpi-3-18.y`

**Comment by CurlyMoo** [comment]
I also made a small mistake. The build folder should be `rpi2-3.18.y` not `rpi2-3-18.y`

File: build/rpi2-3-18.y/config [hunk 3]
```diff
@@ -0,0 +1,7 @@
+config_source_configure=
+config_build_env=CROSS_COMPILE=arm-linux-gnueabihf- ARCH=arm
```
Comment (CurlyMoo) [inline build/rpi2-3-18.y/config]: And still change this `arm-linux-gnueabihf-` to `arm-linux-gnueabi-`

**Comment by f1vefour** [comment]
Done

**Comment by CurlyMoo** [comment]
Tested and confirmed working (building wise).

**Comment by f1vefour** [comment]
Hopefully it boots. You get a Pi 2 yet?

**Comment by CurlyMoo** [comment]
No, and it's already confirmed it doesn't. Seems like a firmware issues rather than a kernel one.

**Comment by f1vefour** [comment]
Shouldn't we be using DTB?

https://github.com/raspberrypi/firmware/tree/master/boot/overlays

**Comment by CurlyMoo** [comment]
As far as i know Raspberry Pi didn't implement `traditional` device trees. I have to see as soon as i get my Raspberry Pi 2.

**Pull request merged**